incubator/hnc/internal/validators/object_test.go

```diff
@@ -308,6 +308,34 @@ func TestUserChanges(t *testing.T) {
 				},
 			},
 		},
+	}, {
+		name: "Deny creation of object with invalid select annotation",
+		fail: true,
+		inst: &unstructured.Unstructured{
+			Object: map[string]interface{}{
+				"apiVersion": "v1",
+				"kind":       "Pod",
+				"metadata": map[string]interface{}{
+					"annotations": map[string]interface{}{
+						api.AnnotationSelector: "$foo" + api.LabelTreeDepthSuffix,
+					},
+				},
+			},
+		},
+	}, {
+		name: "Allow creation of object with valid select annotation",
+		fail: false,
+		inst: &unstructured.Unstructured{
+			Object: map[string]interface{}{
+				"apiVersion": "v1",
+				"kind":       "Pod",
+				"metadata": map[string]interface{}{
+					"annotations": map[string]interface{}{
+						api.AnnotationSelector: "foo" + api.LabelTreeDepthSuffix,
```

Comment:
Ditto here, I'd leave out the tree suffix

Reply:
for the same reason replied above

Reply:
Done